### PR TITLE
Fix building qgis with Qt6 on MinGW-w64

### PR DESCRIPTION
**Pull request by MehdiChinoune**
## Description

These changes fix building QGIS with Qt6 on MinGW-w64

### CODE REVIEW

**Comment by MehdiChinoune** [comment]
@m-kuhn
Just a suggestion, you could install qt6 through https://github.com/jurplel/install-qt-action and install the rest of dependencies using vcpkg to reduce buildtime, unless the dependencies themselves depend on qt6.

File: src/gui/CMakeLists.txt [hunk 1]
```diff
@@ -1664,6 +1664,11 @@ if(MSVC)
     ${CMAKE_BINARY_DIR}/src/gui/qgis_gui_autogen/mocs_compilation.cpp
     PROPERTIES COMPILE_FLAGS "/bigobj"
   )
+elseif(MINGW)
+  set_source_files_properties(
+    ${CMAKE_BINARY_DIR}/src/gui/qgis_gui_autogen/mocs_compilation.cpp
+    PROPERTIES COMPILE_FLAGS "-Wa,-mbig-obj"
```
Comment (m-kuhn) [inline src/gui/CMakeLists.txt:1670]: Is this still required after the recent moc include changes?

**Comment by m-kuhn** [comment]
> @m-kuhn Just a suggestion, you could install qt6 through https://github.com/jurplel/install-qt-action and install the rest of dependencies using vcpkg to reduce buildtime, unless the dependencies themselves depend on qt6.

Dependencies are (including qt) are binary cached, so the build time is most of the time not affected. On the other hand, this requires us to keep custom ports for qtkeychain, qca, ... that consume an external qt version. It also complicates local (dev) builds. (Been there for qfield)

File: src/gui/CMakeLists.txt (hunk 1 above)
Comment (MehdiChinoune) [reply src/gui/CMakeLists.txt:1670]: I don't know, 3.40.0 fails to build on MSYS2/UCRT64.

Comment (MehdiChinoune) [reply src/gui/CMakeLists.txt:1670]: > Is this still required after the recent moc include changes?

which change are you talking about?

Comment (m-kuhn) [reply src/gui/CMakeLists.txt:1670]: https://github.com/qgis/QGIS/pull/59230

**Comment by MehdiChinoune** [comment]
> Dependencies are (including qt) are binary cached, so the build time is most of the time not affected. On the other hand, this requires us to keep custom ports for qtkeychain, qca, ... that consume an external qt version. It also complicates local (dev) builds. (Been there for qfield)
> 
> P.S. I assume there was am update on the windows gh action runner recently, hence the current complete rebuild.

I use vcpkg, It rebuilds all packages if any of them got updated/modified. I recommend you to use vcpkg manifest (https://learn.microsoft.com/en-us/vcpkg/consume/lock-package-versions) if you don't want/need that huge rebuild.

**Comment by m-kuhn** [comment]
> I use vcpkg, It rebuilds all packages if any of them got updated/modified. I recommend you to use vcpkg manifest (https://learn.microsoft.com/en-us/vcpkg/consume/lock-package-versions) if you don't want/need that huge rebuild.

The rebuild only happens when we bump the baseline in vcpkg.json, so not upon an external change.
In any case, I tink this is a good idea for ltr (e.g. keep qt stable while allowing for other patches to land). Do you know if there's something like a `vcpkg freeze` command?

**Comment by MehdiChinoune** [comment]
> Do you know if there's something like a `vcpkg freeze` command?

No, I think the methon I mentioned is the only (supported) way to lock/freeze vcpkg.

**Comment by m-kuhn** [comment]
What I was looking for is a command to help with the maintenance of this. E.g. `pip freeze` offers that and `pip-compile` even more comfortably. IIRC `npm` also has manages the `lock` file with tooling.
Manual file editing for this is prone to be forgotten.

**Comment by m-kuhn** [comment]
Or possibly a better question: which deps should be pinned on which branch and updated at which point in time? I was starting from the principle that we stick to latest on master and bump the baseline occasionally (to be defined) and to older versions in case of troubles. On release branches there are not much thoughts yet (except reacting when something lands on the security list).

**Pull request closed without merging**